src/ReflectionType.php

```diff
@@ -85,14 +85,14 @@ public static function convertToDisplayType(\ReflectionType $type)
             'int'  => 'integer',
             'bool' => 'boolean'
         ];
-        $displayType = $type->type;
+        $displayType = (string)$type;
         if (isset($typeMap[$displayType])) {
             $displayType = $typeMap[$displayType];
         }
 
         $displayType = ltrim($displayType, '\\');
 
-        if ($type->allowsNull()) {
+        if (method_exists($type, 'allowsNull') && $type->allowsNull()) {
```

Comment:
Please return the original code, because Scrutinizer CI issue in https://scrutinizer-ci.com/g/goaop/parser-reflection/inspections/eaf283f6-f982-44a4-bb28-51489bcba537/issues/files/src/ReflectionType.php?status=fixed&orderField=path&order=asc&honorSelectedPaths=0 likely is reported because they're using incorrect PHP stubs for class method signature resolution. On the PHP docs (see http://php.net/manual/en/reflectiontype.allowsnull.php) there is `allowsNull` method and it always was there (as in out `ReflectionType` class polyfill).